### ISSUE TITLE
Support years ranging 0001 to 9999 for Date / Expand range to match other DB (e.g Postgres / Redshift)

### ISSUE THREAD

**Issue by riveraanna** [ClickHouse/ClickHouse#68589]
**Company or project name**

> REA

**Use case**

> We have historical data which requires us to insert dates with year 1800+, but Clickhouse Date, DateTime, and DateTime64 cannot handle a date in the year 1800+. Redshift and Postgres `date` datatype was able to support this.

**Describe the solution you'd like**

> Support years ranging 0001 to 9999 for Date / Expand date range to match other DB (e.g Postgres / Redshift), whichever is feasible.

**Comment by UnamedRus** [comment]
Related issue https://github.com/ClickHouse/ClickHouse/issues/32006

**Comment by zhanglistar** [comment]
Yes, we need enlarge the range for Date. @UnamedRus How do you think?

**Comment by alexey-milovidov** [comment]
Continued in [#81912](https://github.com/ClickHouse/ClickHouse/issues/81912).
